gnat2goto/driver/gnat2goto_itypes.adb

```diff
@@ -190,21 +190,39 @@ package body Gnat2goto_Itypes is
          when N_Integer_Literal => Lower_Bound_Value :=
               Store_Nat_Bound (Bound_Type_Nat (Intval (Lower_Bound)));
          when N_Identifier => Lower_Bound_Value :=
-              Store_Symbol_Bound (Bound_Type_Symbol (Lower_Bound));
+              Store_Symbol_Bound (Bound_Type_Symbol (Do_Identifier
+                                  (Lower_Bound)));
+         when N_Selected_Component => Lower_Bound_Value :=
+              Store_Symbol_Bound (Bound_Type_Symbol (Do_Selected_Component
+                                  (Lower_Bound)));
+         when N_Attribute_Reference => Lower_Bound_Value :=
+              Store_Symbol_Bound (Bound_Type_Symbol (Do_Expression
+                                  (Lower_Bound)));
          when others =>
             Report_Unhandled_Node_Empty (Lower_Bound,
-                                         "Do_Base_Range_Constraint",
+                                         "Do_Modular_Integer",
                                          "unsupported lower range kind");
       end case;
 
       case Nkind (Upper_Bound) is
          when N_Integer_Literal => Upper_Bound_Value :=
               Store_Nat_Bound (Bound_Type_Nat (Intval (Upper_Bound)));
          when N_Identifier => Upper_Bound_Value :=
-              Store_Symbol_Bound (Bound_Type_Symbol (Upper_Bound));
+              Store_Symbol_Bound (Bound_Type_Symbol (Do_Identifier
+                                  (Upper_Bound)));
+         when N_Selected_Component => Upper_Bound_Value :=
+              Store_Symbol_Bound (Bound_Type_Symbol (Do_Selected_Component
+                                  (Upper_Bound)));
+         when N_Attribute_Reference => Upper_Bound_Value :=
+              Store_Symbol_Bound (Bound_Type_Symbol (Do_Expression
+                                  (Upper_Bound)));
+         when N_Op_Add =>
```

Comment:
Why is adding things allowed in the upper bound but not the lower bound? I'm also not completely convinced that special casing all these things for ranges is the right thing to do here